editor/editor_file_system.h

```diff
@@ -235,6 +235,7 @@ class EditorFileSystem : public Node {
 	HashSet<String> textfile_extensions;
 	HashSet<String> valid_extensions;
 	HashSet<String> import_extensions;
+	HashSet<String> processed_files_first_scan;
```

Comment:
The alternative to this annoyingly being a class member is to pass a pointer to a local `HashSet` to `_process_file_system()`, which would only be used in the first scan, which is arguably not much better at all!

Reply:
I like your idea. I added a parameter to `_process_file_system` and pass null when not the first scan.